server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClientImpl.java

```diff
@@ -107,6 +110,23 @@ public ListenableFuture<List<DataSegment>> fetchUsedSegments(String dataSource,
     );
   }
 
+  @Override
+  public ListenableFuture<List<DataSourceInformation>> fetchDataSourceInformation(Set<String> dataSources)
+  {
+    final String path = "/druid/coordinator/v1/metadata/dataSourceInformation";
+    if (null == dataSources) {
+      dataSources = new HashSet<>();
```

Comment:
Why is this null check here?
Should we throw an exception here instead of passing the empty payload to the client?

Reply:
I don't think it is needed. Removed it.